src/arch/mod.rs

```diff
@@ -1,2 +1,5 @@
 #[cfg(target_arch = "x86_64")]
 pub(crate) mod x86_64;
+
+#[cfg(target_arch = "x86_64")]
+pub(crate) use self::x86_64::oops;
```

Comment:
Perhaps we should do a `pub(crate) use self::x86_64::*;` instead, and refer to things which should be provided by any arch as `crate::arch::WHATEVER`.
It would let us avoid `target_arch` checks in other parts of our code.